### PR TITLE
Apply Remove overridden assignment JDT clean in pde.ui

### PR DESCRIPTION
**Pull request by vogella**
Dogfooding the improve JDT cleanup from Bug 572440 to ensure the JDT
cleanup is stable and functional. Push in small chunks to make review
easier.a

### CODE REVIEW

File: ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/editor/contentassist/XMLCompletionProposal.java [hunk 1]
```diff
@@ -329,17 +329,13 @@ private void modifyModel(IDocument document) {
...
 	private void findExtensionVirtualPointValue(IPluginBase base) {
 
-		IDocumentRange range = null;
 		PDESourcePage page = fProcessor.getSourcePage();
 		// Ensure page is defined
 		if (page == null) {
 			return;
 		}
 		// When we inserted the extension element and extension point attribute
-		// name and value, we selected the point value
```
Comment (akurtakov) [inline ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/editor/contentassist/XMLCompletionProposal.java:339]: removed comment, please restore

Comment (vogella) [reply ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/editor/contentassist/XMLCompletionProposal.java:339]: Done

File: ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/editor/contentassist/XMLCompletionProposal.java [hunk 2]
```diff
@@ -329,17 +329,14 @@ private void modifyModel(IDocument document) {
...
 
-		IDocumentRange range = null;
 		PDESourcePage page = fProcessor.getSourcePage();
 		// Ensure page is defined
 		if (page == null) {
 			return;
 		}
 		// When we inserted the extension element and extension point attribute
 		// name and value, we selected the point value
-		// Find the corresponding range in order to add child elements to
```
Comment (akurtakov) [inline ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/editor/contentassist/XMLCompletionProposal.java:340]: You still missed this one.

**Review by HannesWell — APPROVED** [review]
Looks good, besides the two lines Aleks already mentioned.
There is one more comment-line removed in `ContentPage.java` but that's IMHO a trivial comment that is fine to remove.

**Review by HannesWell — APPROVED** [review]
@vogella since you pushed the change directly to the pde-repo I could apply the remarks of Aleks and force pushed the updated commit (also rebased on top of master).

**Comment by vogella** [comment]
How smart ot me, to accidently push directly to this repo instead of my vogellacaompany clone.............. :-o (just kidding)

Thanks, @HannesWell for updating my PR. @akurtakov if you OK with the change, please merge

**Review by akurtakov — APPROVED** [review]

**Comment by vogella** [comment]
Thanks again @HannesWell and also thanks to @akurtakov

**Comment by HannesWell** [comment]
> How smart ot me, to accidently push directly to this repo instead of my vogellacaompany clone.............. :-o (just kidding)
> 
Very smart 😄
You are welcome and thanks for the change.

Btw. do you know who is mostly improving the clean-ups? I have some suggestions for clean-ups that could be beneficial for the equinox and pde code-base in many places:
- [Bug 575078](https://bugs.eclipse.org/bugs/show_bug.cgi?id=575078)
- [Bug 575008](https://bugs.eclipse.org/bugs/show_bug.cgi?id=575008)
- [Bug 575009](https://bugs.eclipse.org/bugs/show_bug.cgi?id=575009)

Unfortunately I'm not familiar with the JDT-code and at the moment have no time to work on that.

**Comment by akurtakov** [comment]
Feel free to talk to @jjohnstn - he has been the one enhancing and reviewing others cleanups lately.

**Comment by HannesWell** [comment]
> Feel free to talk to @jjohnstn - he has been the one enhancing and reviewing others cleanups lately.

Thank you.

**Comment by vogella** [comment]
@HannesWell we already have a ton of exsting cleanups, e.g. for performance. Would be great, if you can also apply the once you find useful also to platform, Equinox and PDE and report any issues you find with them also to JDT UI.

**Comment by HannesWell** [comment]
> @HannesWell we already have a ton of exsting cleanups, e.g. for performance. Would be great, if you can also apply the once you find useful also to platform, Equinox and PDE and report any issues you find with them also to JDT UI.

Yes, I have the intention to do that it is just a big task and I have other things with higher priority on my list. But I already applied some and will do more. It just takes time. If anybody else applies the clean ups too, it would be of course welcome. :)

**Comment by jjohnstn** [comment]
I have assigned the convert while loop to enhanced for loop cleanup to myself and asked Thomas to take a look at the other two.

**Comment by HannesWell** [comment]
> I have assigned the convert while loop to enhanced for loop cleanup to myself and asked Thomas to take a look at the other two.

Thanks!



> @HannesWell we already have a ton of exsting cleanups, e.g. for performance. Would be great, if you can also apply the once you find useful also to platform, Equinox and PDE and report any issues you find with them also to JDT UI.

Btw. [Bug 575008](https://bugs.eclipse.org/bugs/show_bug.cgi?id=575008) and [Bug 575009](https://bugs.eclipse.org/bugs/show_bug.cgi?id=575009) are a result of applying the clean-up to the Equinox code-base. :)

**Pull request merged**